core/camel-support/src/main/java/org/apache/camel/support/AbstractExchange.java

```diff
@@ -189,7 +189,15 @@ public Exchange copy() {
 
     @SuppressWarnings("unchecked")
     private void safeCopyProperties(Map<String, Object> source, Map<String, Object> target) {
-        target.putAll(source);
+        source.entrySet().stream().forEach(entry -> {
```

Comment:
Sorry we do not want this kind in the core. Can you maybe try to come up with a different solution.

Reply:
@davsclaus Are you suggesting to totally avoid changes in core, or avoid these specific changes in core? I initially did come up with another approach for camel-zipkin but it is far more complicated and honestly does not address the real problem.  

The span problem I am pretty sure exists in camel opentracing and opentelemetry as well because both of them use camel-tracing and if you look at [ActiveSpanManager](https://github.com/apache/camel/blob/master/components/camel-tracing/src/main/java/org/apache/camel/tracing/ActiveSpanManager.java) it also uses Exchange property mechanism to set  ACTIVE_SPAN_PROPERTY which would get corrupted when using parallelProcessing with multicast. 

My suggested approach would easily fix issues in opentracing and opentelemetry as well.

Reply:
Yes we do not want core comprimised with this due to zipkin storing state like that. The core should be lean and fast.

Reply:
Instead of doing this expensive check always, then find out some way for zipkin et all to turn on a flag on the exchange (via ExtendedExchange) to tell it that it should do deep clone safe copy mode, then this is only in use when you use these special zipkin components.

Or instead of storing the zipkin state as exchange property, store it on some zipkin bean, that handles this and stores the data per exchange id or something.

Reply:
@davsclaus Thanks for the advice. I think setting the flag via ExtendedExchange would be the best option.

Reply:
I have incorporated your feedback and also an alternate approach and IMHO cleaner one has been implemented in another PR https://github.com/apache/camel/pull/5373

Reply:
Oh that is great work that you implemented two different solutions. I agree with you that the alternative is the better choice, so lets close this PR